internal/services/maintenance/public_maintenance_configurations_data_source.go

```diff
@@ -0,0 +1,191 @@
+
+			"location_filter": {
+				Type:      pluginsdk.TypeString,
+				Optional:  true,
+				StateFunc: azure.NormalizeLocation,
+			},
+
+			"scope_filter": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"All", // All is still accepted by the API
+					string(maintenance.ScopeExtension),
+					string(maintenance.ScopeHost),
+					string(maintenance.ScopeInGuestPatch),
+					string(maintenance.ScopeOSImage),
+					string(maintenance.ScopeSQLDB),
+					string(maintenance.ScopeSQLManagedInstance),
+				}, false),
+			},
+
+			"recur_every_filter": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					recurMondayToThursday,
+					recurFridayToSunday,
+				}, false),
+			},
+
+			"public_maintenance_configurations": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"location": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"duration": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"maintenance_scope": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"time_zone": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"recur_every": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourcePublicMaintenanceConfigurationsRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Maintenance.PublicConfigurationsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resp, err := client.List(ctx)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("no Public Maintenance Configurations were found")
+		}
+		return fmt.Errorf("retrieving Public Maintenance Configurations: %+v", err)
+	}
+
+	filteredPublicConfigs := make([]interface{}, 0)
+
+	recurEveryFilter := d.Get("recur_every_filter").(string)
+	if recurEveryFilter == recurFridayToSunday {
+		recurEveryFilter = "week Friday, Saturday, Sunday"
+	} else if recurEveryFilter == recurMondayToThursday {
+		recurEveryFilter = "week Monday, Tuesday, Wednesday, Thursday"
+	}
+
+	locationFilter := d.Get("location_filter").(string)
+	scopeFilter := d.Get("scope_filter").(string)
+
+	for _, maintenanceConfig := range *resp.Value {
+
+		var configLocation, configRecurEvery, configScope string
+		if maintenanceConfig.Location != nil {
+			configLocation = *maintenanceConfig.Location
```

Comment:
the location here wants to be normalized fwiw (e.g. `West Europe` -> `westeurope`)